server/src/main/java/org/opensearch/cluster/routing/allocation/decider/ThrottlingAllocationDecider.java

```diff
@@ -357,7 +357,7 @@ private ShardRouting initializingShard(ShardRouting shardRouting, String current
     @Override
     public Decision canMoveAway(ShardRouting shardRouting, RoutingAllocation allocation) {
         int outgoingRecoveries = 0;
-        if (!shardRouting.primary()) {
+        if (!shardRouting.primary() && !shardRouting.isSearchOnly()) {
```

Comment:
Sorry couldn't understand the context for this change. This is meant to move the replica and check if the outgoing recovery from it's peer primary is getting throttled. Wouldn't search only replica recover from the primary copy or the plan is to make recoveries totally independent? cc: @mch2

Reply:
Err thanks for calling this out I'm not handling this correclty.

This was to resolve an NPE a few lines later when we only have search replicas and the primary has a node drop. In this case the NPE occurs right after this where we fetch the active primary as null and then invoke .currentNodeId on it.  We should be setting outgoingRecoveries to 0 in this case instead.

To answer your question the plan is to have search replica recover directly from the remote store when its enabled but we need to still honor node-node with active primary.